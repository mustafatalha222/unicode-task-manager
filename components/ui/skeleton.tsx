import { cn } from '@/lib/utils'
import { useTranslations } from 'next-intl'

function Skeleton({ className, ...props }: React.HTMLAttributes<HTMLDivElement>) {
  return <div className={cn('animate-pulse rounded-md bg-primary/10', className)} {...props} />
}

function CustomSkeleton({
  className,
  loading = false,
  length = 0,
  ...props
}: React.HTMLAttributes<HTMLDivElement> & { loading: boolean; length: number }) {
  const t = useTranslations()
  return (
    <>
      {loading && !length ? (
        <div className={cn('animate-pulse rounded-md bg-primary/10', className)} {...props} />
      ) : (
        !length && <h4 className="text-xl font-bold flex justify-center">{t('No Data Found')}</h4>
      )}
    </>
  )
}

export { Skeleton, CustomSkeleton }
