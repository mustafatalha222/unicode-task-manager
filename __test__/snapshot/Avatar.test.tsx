import { Avatar, AvatarFallback, AvatarImage } from '@/components/ui/avatar'
import renderer from 'react-test-renderer'

it('renderer language Switcher unchange', () => {
  const tree = renderer
    .create(
      <Avatar>
        <AvatarImage src={''} />
        <AvatarFallback>IM</AvatarFallback>
      </Avatar>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
