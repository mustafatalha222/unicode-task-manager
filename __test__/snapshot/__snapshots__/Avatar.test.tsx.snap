// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer language Switcher unchange 1`] = `
<span
  className="relative flex h-10 w-10 shrink-0 overflow-hidden rounded-full"
>
  <span
    className="flex h-full w-full items-center justify-center rounded-full bg-muted"
  >
    IM
  </span>
</span>
`;
